tests/test_compilation.py

```diff
@@ -37,7 +37,7 @@ def logprob_fn(x):
     def test_nuts(self):
         # Log probability function was traced twice as we call it
         # at Step 0 when building a new trajectory in tree doubling.
-        @chex.assert_max_traces(n=2)
+        @chex.assert_max_traces(n=1)
```

Comment:
I like that! Any idea of the performance difference:
- For small `n_samples` where time is dominated by compilation
- For very large `n_samples` where time is dominated by running time of the jitted kernel

Reply:
I think we should benchmark on a simple example like the linear regression in the tests to see if the change is overall better (decreases compile time & execution time stays the same) or if there is a trade-off.